packages/test/stochastic-test-utils/src/performActions.ts

```diff
@@ -99,15 +100,16 @@ export async function performFuzzActionsAsync<
 	const applyOperation: (operation: TOperation) => Promise<TState> = async (op) =>
 		(await reducer(state, op)) ?? state;
 
+	if (saveInfo?.filepath !== undefined) {
+		await fs.mkdir(path.dirname(saveInfo.filepath), { recursive: true });
+	}
```

Comment:
This goes against the contract advertised on `SaveInfo.filepath`. It would also result in the folder being created even when no file will be written to it, which not a big problem but can be an annoyance sometimes.

I'm leaning towards either keeping the contract as it was or automatically creating the folder when we actually want to write a file to it (we'd probably put all the folder/file creation code in a pair of sync/async functions for terseness). @Abe27342 do you have any thoughts this?

Reply:
i think second option there is best (automatically create the folder when about to write). perf of checking the containing path exists repeatedly for runs with lots of failures isn't really a concern here, and it removes some boilerplate from callers.